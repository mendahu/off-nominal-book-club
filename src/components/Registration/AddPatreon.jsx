import { Typography, Button } from '@material-ui/core'
import Layout from '../DefaultLayout'
import patreonAuthUrlGenerator from '../../../src/helpers/patreon/authUrlGenerator'

const patreonAuthOptions = {
  client_id: process.env.PAT_CLIENT_ID,
  redirect_uri: process.env.PAT_REDIRECT_URI
}

export default function AddPatreon() {
  return (
    <Layout>
      <Typography>We just need a little more info!</Typography>
      <Typography>Connect your Patreon account to be able to rate and review books, as well as add new books to the app.</Typography>
      <Button variant="contained" color="secondary" href={patreonAuthUrlGenerator(patreonAuthOptions)}>Connect your Patreon</Button>
    </Layout>
  )
}