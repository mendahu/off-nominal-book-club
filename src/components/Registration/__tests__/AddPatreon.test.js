import { shallow } from 'enzyme';
import { Button } from '@material-ui/core';
import AddPatreon from '../AddPatreon';
import axios from 'axios';
jest.mock('axios');

describe('Add Patreon', () => {
  it('should render and match snapshot', () => {
    const wrapper = shallow(<AddPatreon />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should execute skipProfile if skip button clicked', () => {
    const mockSkipProfile = jest.fn();
    axios.post.mockImplementation(() => 'post');

    const wrapper = shallow(<AddPatreon skipProfile={mockSkipProfile} />);
    wrapper.find(Button).at(3).simulate('click');
    expect(mockSkipProfile).toHaveBeenCalledTimes(1);
    expect(axios.post).toHaveBeenCalledTimes(1);
  });
});
