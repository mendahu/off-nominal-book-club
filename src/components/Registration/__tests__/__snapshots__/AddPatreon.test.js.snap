// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Patreon should render and match snapshot 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-root-2 makeStyles-centred-1"
>
  <img
    alt="Patreon Logo"
    className="makeStyles-patreonWordMark-4 makeStyles-centred-1"
    src="/Patreon_White.png"
  />
  <WithStyles(ForwardRef(Typography))
    variant="body1"
  >
    Patrons of the WeMartians Podcast and Main Engine Cut Off Podcast have special privileges at the Off-Nominal Book Club. All Patrons can:
  </WithStyles(ForwardRef(Typography))>
  <ul>
    <WithStyles(ForwardRef(Typography))
      component="li"
      variant="body1"
    >
      Add books to the collection
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="li"
      variant="body1"
    >
      Rate books
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="li"
      variant="body1"
    >
      Review books
    </WithStyles(ForwardRef(Typography))>
  </ul>
  <WithStyles(ForwardRef(Typography))
    align="center"
    className="makeStyles-topSpace-5"
    component="p"
    variant="h5"
  >
    Already a patron?
  </WithStyles(ForwardRef(Typography))>
  <div
    className="makeStyles-centredText-7"
  >
    <WithStyles(ForwardRef(Button))
      className="makeStyles-button-3"
      color="primary"
      endIcon={<Memo />}
      href="https://www.patreon.com/oauth2/authorize?response_type=code&client_id=undefined&redirect_uri=undefined&scope=identity identity.memberships"
      variant="contained"
    >
      Connect Patreon
    </WithStyles(ForwardRef(Button))>
  </div>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-topSpace-5"
    paragraph={true}
    variant="h5"
  >
    Not a patron?
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    paragraph={true}
    variant="body1"
  >
    Consider supporting us for as little as $1/month. Not only do you get access to the premium features of Off-Nominal Book Club, but you support two independent content creators helping tell the story of space exploration. Patrons get many other perks as well!
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-centredText-7"
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-3"
        color="secondary"
        endIcon={<Memo />}
        href="http://www.wemartians.com/support"
        variant="contained"
      >
        Support WeMartians
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-centredText-7"
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-3"
        color="secondary"
        endIcon={<Memo />}
        href="http://www.mainenginecutoff.com/support"
        variant="contained"
      >
        Support MECO
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-button-3 makeStyles-topSpace-5"
    color="default"
    endIcon={<Memo />}
    onClick={[Function]}
    variant="contained"
  >
    Skip for now
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-paragraphSpace-6"
    component="p"
    variant="caption"
  >
    Don't worry - free users can still browse the collection, mark books as read or favourites, and create wishlists.
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Paper))>
`;
